rosidl_typesupport_opendds_cpp/include/rosidl_typesupport_opendds_cpp/service_type_support.h

```diff
@@ -18,31 +18,45 @@
 #include <stdint.h>
 #include <rmw/types.h>
 #include "rosidl_generator_c/service_type_support_struct.h"
+#include "dds/DCPS/Service_Participant.h"
+
+typedef void* (*allocator_t)(size_t);
+typedef void (*deallocator_t)(void*);
 
 typedef struct service_type_support_callbacks_t
 {
   const char * service_namespace;
   const char * service_name;
-  // Function to create a requester
+  //! Function to create a requester
+  /*!
+  Set pub/sub default writer/reader qos to the ros converted qos, before calling this function
+  TODO: add 'const' to parameters, when determined they stay unchanged by the Requestor
+  */
   void * (*create_requester)(
-    void * participant,
+    DDS::DomainParticipant_var dds_participant,
     const char * request_topic_str,
     const char * response_topic_str,
-    const void * datareader_qos, const void * datawriter_qos,
-    void ** reader, void ** writer, void * (*allocator)(size_t));
-  // Function to destroy a requester
+    DDS::Publisher_var dds_publisher,
+    DDS::Subscriber_var dds_subscriber,
+    allocator_t allocator);
+  //! Function to destroy a requester
   const char * (*destroy_requester)(
-    void * untyped_requester, void (* deallocator)(void *));
-  // Function to create a replier
+    void * untyped_requester, deallocator_t deallocator);
+  //! Function to create a replier
+  /*!
+  Set pub/sub default writer/reader qos to the ros converted qos, before calling this function
+  TODO: add 'const' to parameters, when determined they stay unchanged by the Requestor
+  */
   void * (*create_replier)(
-    void * participant,
-    const char * request_topic_str,
-    const char * response_topic_str,
-    const void * datareader_qos, const void * datawriter_qos,
-    void ** reader, void ** writer, void * (*allocator)(size_t));
-  // Function to destroy a replier
+    DDS::DomainParticipant_var dds_participant,
+    const char* request_topic_str,
```

Comment:
Determine what our "Style Guide" is, and be consistent `const char* s` vs `const char * s` (I don't like the latter since it looks like binary operator).

Reply:
I agree with your preference but the rest of code uses other style, so I'll keep its style.

Reply:
As long as we're consistent (especially within the same file), I'm OK with it

Reply:
It's preferable just to use the tools that ROS2 already offers. 

https://answers.ros.org/question/306194/ros1-and-ros2-linting/

Reply:
Should that be in the CI scripts?

Reply:
As a check, I guess so. I'd have to look into it more. But code beautifiers would be required to run prior to commit.

Reply:
Right, I meant as a check -- just like we do in OpenDDS for some things

Reply:
Following the link Jeremy provided, I found some tools for checking the source code: cppcheck, and astyle. You can specify where the * or & go (left, middle, or right). However, I still can't find the ros style guidelines. If there is none on this particular point, we may as well use our OpenDDS guideline, at least consistent with ourselves.

Reply:
At this point for this PR, I think we should table this particular issue. This is the only remaining issue for this PR and I don't think it requires action to accept the PR. We definitely want a consistent style. There are tools to do so. We'll get there.